tsl/test/expected/cagg_planning.out

```diff
@@ -0,0 +1,170 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+\set EXPLAIN 'EXPLAIN (costs off, timing off)'
+SET timezone TO PST8PDT;
+CREATE TABLE metrics(time timestamptz, device text, metric text, value float);
+SELECT create_hypertable('metrics', 'time');
+NOTICE:  adding not-null constraint to column "time"
+  create_hypertable   
+----------------------
+ (1,public,metrics,t)
+(1 row)
+
+INSERT INTO metrics SELECT '2020-01-01'::timestamptz + format('%s day', i::text)::interval, 'device 1', 'metric 1', i  FROM generate_series(0, 9, 1) g(i);
+-- cagg with grouping only by time column
+CREATE MATERIALIZED VIEW cagg1 WITH (timescaledb.continuous,timescaledb.materialized_only=false)
+AS SELECT time_bucket('3 day', time), avg(value) FROM metrics GROUP BY 1;
+NOTICE:  refreshing continuous aggregate "cagg1"
+-- cagg with grouping by device and time column
+CREATE MATERIALIZED VIEW cagg2 WITH (timescaledb.continuous,timescaledb.materialized_only=false)
+AS SELECT device, time_bucket('3 day', time), avg(value) FROM metrics GROUP BY device, 2;
+NOTICE:  refreshing continuous aggregate "cagg2"
+INSERT INTO metrics SELECT '2020-01-01'::timestamptz + format('%s day', i::text)::interval, 'device 1', 'metric 1', i  FROM generate_series(10, 16, 1) g(i);
+:EXPLAIN SELECT * FROM cagg1 ORDER BY 1;
+                                                      QUERY PLAN                                                       
+-----------------------------------------------------------------------------------------------------------------------
+ Append
```

Comment:
This looks incorrect, an Append node is not guaranteed to produce ordered output, so it can't be on top when there's an ORDER BY.

Reply:
This is the idea of ordered append, when ordering by time and under the assumption that the individual children are ordered you can just append individual children and preserve ordering

Reply:
the first child is the materialized cagg and the 2nd child is the realtime-part, the materialized part will always be ordered before the realtime-part when ordering by time.

Reply:
Should it have pathkeys then? Looks like it doesn't, no "Order" in explain unlike the below nodes.

Reply:
Hmm atm it doesnt but it would be useful if it had them if this was a subquery in a bigger query. I removed the pathkeys from that level because postgres will turn it into mergeappend which means it will execute all children and prevent early abort with LIMIT

Reply:
postgres is not aware that the children are already ordered based on the position in the append so it puts a mergeappend there cause its only aware of the local sorting not the global sorting. path creation is done by postgres here and we only amend the initial parse tree a bit before actual path generation